### DISCUSSION TITLE
Protecting custom config names from future name-collision

### DISCUSSION THREAD

**Discussion by szabgab** [rwf2/Rocket#2724] [Questions]
If I understand correctly the internal configuration parameters of Rocket and any custom configuration live in the same name-space. So if and when a new configuration parameter is added to Rocket it might collide with a custom parameter in a project, right?

If that's the case is there a recommended way to avoid it?

I guess prefixing my custom parameters with the name of the project would reduce the chances to almost 0, right?

**Comment by edgarogh** [comment]
Rocket uses [`figment`](https://github.com/SergioBenitez/Figment) to parse its config, and I'm pretty sure you don't really have a way to detect such collisions. It's also explicity mentioned in the source code that Rocket's native config ["may grow in a non-breaking release"](https://docs.rs/rocket/0.5.0/src/rocket/config/config.rs.html#122-124).

If I were you and I was worried about collisions, I'd indeed prefix it. If you still don't want to do that, just be careful when you run `cargo update` and always check that the [`Config`](https://docs.rs/rocket/latest/rocket/config/struct.Config.html) struct hasn't changed. You can also write unit/integration tests as a safety net.
